lib/tlIO/OpenEXRRead.cpp

```diff
@@ -153,76 +154,87 @@ namespace tl
                     {
                         _s.reset(new IStream(fileName.c_str()));
                     }
-                    _f.reset(new Imf::InputFile(*_s));
+                    _f.reset(new Imf::MultiPartInputFile(*_s));
+                    int numberOfParts = _f->parts();
+                    int partNumber;
 
-                    // Get the display and data windows.
-                    _displayWindow = fromImath(_f->header().displayWindow());
-                    _dataWindow = fromImath(_f->header().dataWindow());
-                    _intersectedWindow = _displayWindow.intersect(_dataWindow);
-                    _fast = _displayWindow == _dataWindow;
-
-                    if (auto logSystem = logSystemWeak.lock())
+                    for ( partNumber = 0; partNumber < numberOfParts; ++partNumber )
                     {
-                        const std::string id = string::Format("tl::io::exr::Read {0}").arg(this);
-                        std::vector<std::string> s;
-                        s.push_back(string::Format(
-                            "\n"
-                            "    file name: {0}\n"
-                            "    display window {1}\n"
-                            "    data window: {2}\n"
-                            "    compression: {3}").
-                            arg(fileName).
-                            arg(_displayWindow).
-                            arg(_dataWindow).
-                            arg(getLabel(_f->header().compression())));
-                        const auto& channels = _f->header().channels();
-                        for (auto i = channels.begin(); i != channels.end(); ++i)
-                        {
-                            std::stringstream ss2;
-                            ss2 << "    channel " << i.name() << ": " << getLabel(i.channel().type) << ", " << i.channel().xSampling << "x" << i.channel().ySampling;
-                            s.push_back(ss2.str());
-                        }
-                        logSystem->print(id, string::join(s, '\n'));
-                    }
+                        const Imf::Header& header = _f->header(partNumber);
 
-                    // Get the tags.
-                    readTags(_f->header(), _info.tags);
+                        // Get the display and data windows.
+                        _displayWindow = fromImath( header.displayWindow());
```

Comment:
All of these (display window, data window, etc.) are the same across all parts?

Reply:
Yes.